### PR TITLE
Shell support

### PR DESCRIPTION
**Pull request by kmokstad**
This will add a new integrand class for linear Kirchhoff-Love shells.

The first commit introduces the intermediate class KirchhoffLove which serves as a common base for the present KirchhoffLovePlate and the new KircjjoffLoveShell class.

Currently only the framework, the integrand implementations is to be provided through a Masters project during spring 2018.

### CODE REVIEW

**Comment by kmokstad** [comment]
jenkins build this with downstreams please

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem=253 please

**Comment by akva2** [comment]
faulty commit msg in last commit.

File: Linear/main_LinEl.C [hunk 1]
```diff
@@ -67,6 +68,7 @@
   \arg -2Dpstrain : Use two-parametric simulation driver (plane strain)
   \arg -2Daxisymm : Use two-parametric simulation driver (axi-symmetric solid)
   \arg -2DKL : Use two-parametric simulation driver for Kirchhoff-Love plate
+  \arg -2DKLshel : Use two-parametric simulation driver for Kirchhoff-Love shell
```
Comment (akva2) [inline Linear/main_LinEl.C]: l

**Comment by kmokstad** [comment]
No longer in progress. Ready for final review now.

**Pull request closed without merging**